src/Controls/src/Core/Element.cs

```diff
@@ -97,7 +98,99 @@ public string StyleId
 			}
 		}
 
-		internal virtual IReadOnlyList<Element> LogicalChildrenInternal => EmptyChildren;
+		// Leaving this internal for now.
+		// If users want to add/remove from this they can use
+		// AddLogicalChildren and RemoveLogicalChildren on the respective control
+		// if available.
+		//
+		// Ultimately I don't think we'll need these to be virtual but some controls (layout)
+		// are going to take a more focused effort so I'd rather just do that in a 
+		// separate PR. I don't think there's ever a scenario where a subclass needs
+		// to replace the backing store.
+		// If everyone just uses AddLogicalChildren and RemoveLogicalChildren
+		// and then overrides OnChildAdded/OnChildRemoved
+		// that should be sufficient
+		internal IReadOnlyList<Element> LogicalChildrenInternal
+		{
+			get
+			{
+				SetupChildren();
+				return _logicalChildrenInternal;
+			}
+		}
+
+		private protected virtual IList<Element> LogicalChildrenInternalBackingStore
+		{
+			get
+			{
+				_internalChildren ??= new List<Element>();
+				return _internalChildren;
+			}
+		}
+
+
+		void SetupChildren()
+		{
+			_logicalChildrenInternal ??= new ReadOnlyCollection<Element>(LogicalChildrenInternalBackingStore);
+		}
+
+		internal void InsertLogicalChildInternal(int index, Element element)
```

Comment:
Should these rather be public so toolkits can add/remove things from the logical tree? Do we want to support that? You hint at this for `Page` in the description.

Reply:
Yea, that's where I started with this PR was making these public. Then I just got a little worried about having these on everything (MenuItems, Layouts, etc...), so I put that on the `will figure out later` pile. Once folks are happy with this PR and we merge it, then I'll do a more targeted PR around what it would look like to make these public.  Do we only want them public on `Page`? Do we want them public on all `VisualElement`? Would it be weird/useful having an API on `VerticalStackLayout` where a user could just add random `LogicalChildren`?